h2o-core/src/main/java/hex/ModelMojoWriter.java

```diff
@@ -80,12 +82,39 @@ public ModelMojoWriter(M model) {
     }
   }
 
+  @Override
+  protected void writeModelKV() throws IOException {
+    writeCommonModelData();
+    writeModelData();
+  }
+  
+  protected void writeCommonModelData() throws IOException {
+    if (model._parms._preprocessors != null) {
+      writekv("preprocessors_count", model._parms._preprocessors.length);
+    }
+  }
+  
+  protected abstract void writeModelData() throws IOException;
+
   @Override
   protected void writeExtraInfo() throws IOException {
+    super.writeExtraInfo();
     writeModelDetails();
     writeModelDetailsReadme();
+    writeModelPreprocessors();
+  }
+  
+  protected void writeModelPreprocessors() throws IOException {
+    if (model._parms._preprocessors == null) return;
+    for (int i=0; i < model._parms._preprocessors.length; i++) {
+      startWritingTextFile("experimental/preprocessor_"+i);
```

Comment:
no use using experimental dir name, this will aways forever stay the same, can be just `preprocessors/preprocessor_i`

Reply:
good point, was actually wondering about this as changing the path later is a source for future backwards incompatibility, will change.